api/api/services/internal_tasks/integration_service.py

```diff
@@ -0,0 +1,335 @@
+import datetime
+import logging
+from enum import Enum
+from typing import AsyncIterator, Literal, NamedTuple
+
+from pydantic import BaseModel, Field
+
+from api.services.api_keys import APIKeyService
+from api.services.runs import RunsService
+from core.agents.agent_name_suggestion_agent import (
+    AGENT_ID as AGENT_NAME_SUGGESTION_AGENT_ID,
+)
+from core.agents.agent_name_suggestion_agent import (
+    AgentNameSuggestionAgentInput,
+    agent_name_suggestion_agent,
+)
+from core.agents.integration_agent import (
+    IntegrationAgentChatMessage,
+    IntegrationAgentInput,
+    integration_chat_agent,
+)
+from core.domain.agent_run import AgentRun
+from core.domain.errors import ObjectNotFoundError
+from core.domain.events import EventRouter
+from core.domain.integration_domain import (
+    OFFICIAL_INTEGRATIONS,
+    PROPOSED_AGENT_NAME_PLACEHOLDER,
+    WORKFLOWAI_API_KEY_PLACEHOLDER,
+    Integration,
+    IntegrationKind,
+)
+from core.domain.task_variant import SerializableTaskVariant
+from core.domain.users import User
+from core.storage.backend_storage import BackendStorage
+from core.utils.redis_cache import redis_cached
+
+DEFAULT_AGENT_ID = "default"
```

Comment:
@guillaq ideally this should be propagated everywhere.

Reply:
https://github.com/WorkflowAI/WorkflowAI/pull/247/commits/20a2b9fbf31dd0a7f9ac6dd6e9436bb14a789ddd